### PR TITLE
Upgrade AllDay PackNFT Smart Contract

### PR DESCRIPTION
**Pull request by judezhu**
1. Use resource UUID as NFT flow id, instead of using incremental integer
2. Added `Burned` Event.
3. Upgrade `IPackNFT` to be compatible with AllDay PackNFT contract

### CODE REVIEW

File: pds/contracts/PackNFT_AllDay.cdc [hunk 1]
```diff
@@ -125,8 +125,17 @@ pub contract PackNFT: NonFungibleToken, IPackNFT {
             PackNFT.openRequest(id: self.id)
         }
 
-        init(initID: UInt64, commitHash: String, issuer: Address ) {
-            self.id = initID
+       destroy() {
+           let p <- PackNFT.packs.remove(key: self.id) ?? panic("no such pack")
+           PackNFT.totalSupply = PackNFT.totalSupply - (1 as UInt64)
```
Comment (Deewai) [inline pds/contracts/PackNFT_AllDay.cdc:130]: I know we no longer use this to calculate the id but to be on a safer side, let's not decrement it

**Review by Deewai — APPROVED** [review]
LGTM!

File: pds/contracts/IPackNFT.cdc [hunk 2]
```diff
@@ -26,10 +26,6 @@ pub contract interface IPackNFT{
     /// This is emitted when owner of a PackNFT request for the entitled NFT to be
     /// deposited to its account
     pub event OpenRequest(id: UInt64)
-    /// New Pack NFT
```
Comment (sadief) [inline pds/contracts/IPackNFT.cdc:29]: So there would be no Minted event? 🤔

Comment (judezhu) [reply pds/contracts/IPackNFT.cdc:29]: The issue was we made some "reckless" [changes](https://github.com/dapperlabs/platform-services/pull/599/) in the past, to make PackNFT to be more aligned with freshmint BlindNFT. But these changes make AllDay PackNFT in-compatible with our PDS anymore. After a discussion, we decided to remove the events and fields in IPackNFT to make AllDay PackNFT contract to be compatible. This is not the perfect solution. But the best we have right now.

Comment (Deewai) [reply pds/contracts/IPackNFT.cdc:29]: Minted event will still remain in the `PackNFT` contract. All events and fields removed from the `IPackNFT` will still remain in the `PackNFT`

File: pds/contracts/PackNFT_AllDay.cdc (hunk 1 above)
Comment (Deewai) [reply pds/contracts/PackNFT_AllDay.cdc:130]: okay, we can leave it then

File: pds/contracts/IPackNFT.cdc (hunk 2 above)
Comment (sadief) [reply pds/contracts/IPackNFT.cdc:29]: gotcha 👍  makes sense!

**Review by sadief — APPROVED** [review]

**Pull request merged**